### ISSUE TITLE
API Proposal: Signaling

### ISSUE THREAD

**Issue by msach22** [opentok/OpenTok-PHP-SDK#208]
This is an API Proposal regarding server side signaling.

## Signal
The platform allows a signal to be sent to an active session. See [docs](https://tokbox.com/developer/rest/#send_signal).

We propose to add a new method for sending a signal. 

### `signal()`
```php

# To send a signal to everyone in the session:
$options = array(
  'type' => 'signal type', // string (optional)
  'data' => 'signal data', // string (optional)
);
$opentok->signal($sessionId, $options);

# Send a signal from an Session instance (returned from createSession)
$session->signal($options);

# To send a signal to a specific connection in the session:
$options = array(
  'type' => 'signal type', // string (optional)
  'data' => 'signal data' // string (optional)
);
$connectionId = '2a84cd30-3a33-917f-9150-49e454e01572',

$opentok->signal($sessionId, $options, $connectionId);

# Send a signal from an Session instance (returned from createSession)
$session->signal($options);
```

**Comment by msach22** [comment]
@jeffswartz Do you think it's better to have the connection Id be a part of the `$options` associative array or be it's own parameter and evolve the respective methods to the following:

`$opentok->signal($sessionId, $options, $connectionId);`
`$session->signal($options, $connectionId);`

**Comment by jeffswartz** [comment]
To be consistent with the Node.js SDK, let's name the parameter `$payload`, not `$options`.

**Comment by jeffswartz** [comment]
@msach22 please add details on error conditions and Error classes being added.

**Comment by msach22** [comment]
Released with 4.2.0
